app/utils/BFS.js

```diff
@@ -1,122 +1,120 @@
-		}
-	}
+  // tests if the tested point is in fact the ending point of the search
+  const isEndPoint = function (point) {
+    if (point === endP) {
+      return true;
+    }
+    return false;
+  };
 
-	// returns the neighbours of a certain point
-	var getNeighbours = function (point) {
-		var neighbours = [];
-		var movementX = [1, -1, 0, 0];
-		var movementY = [0, 0, 1, -1];
-		for (var i=0; i<4; i++) {
-			let neighbour = [];
-			neighbour[0] = point[0] + movementY[i];
-			neighbour[1] = point[1] + movementX[i];
-			neighbours.push(neighbour);
-		}
-		return neighbours;
-	}
+  // returns the neighbours of a certain point
+  const getNeighbours = function (point) {
+    const neighbours = [];
+    const movementX = [ 1, -1, 0, 0, ];
+    const movementY = [ 0, 0, 1, -1, ];
+    for (let i = 0; i < 4; i++) {
+      const neighbour = [];
+      neighbour[0] = point[0] + movementY[i];
+      neighbour[1] = point[1] + movementX[i];
+      neighbours.push(neighbour);
+    }
+    return neighbours;
+  };
 
-	// tests if a certain point can be used (inside the board && allowed to be stepped in && was not visited)
-	var legalPoint = function (point) {
-		if (inField(point)) {
-			if (allowedStep(point)) {
-				if (!visited(point)) {
-					return true;
-				}
-			}
-		}
-		return false;
-	}
+  // tests if a certain point can be used (inside the board && allowed to be stepped in && was not visited)
+  const legalPoint = function (point) {
+    if (inField(point)) {
+      if (allowedStep(point)) {
+        if (!visited(point)) {
+          return true;
+        }
+      }
+    }
+    return false;
+  };
 
-	// returns the neighbours that are valid points to be used
-	var legalNeighbours = function (neighbours) {
-		var legal = [];
-		neighbours.forEach(function(neighbour) {
-			if (legalPoint(neighbour)) {
-				legal.push(neighbour);
-			}
-		})
-		return legal;
-	}
+  // returns the neighbours that are valid points to be used
+  const legalNeighbours = function (neighbours) {
+    const legal = [];
+    neighbours.forEach((neighbour) => {
+      if (legalPoint(neighbour)) {
+        legal.push(neighbour);
+      }
+    });
+    return legal;
+  };
 
-	// the BFS algorithm that searches for a path between the 'start' and 'end' points
-	// returns 'true' if a path was found and 'false' if not
-	var pathExists = function() {
-		if (legalPoint(startP) && legalPoint(endP)) {
-			var testingQue = [startP];	
-			field[testingQue[0][0]][testingQue[0][1]].stepVisited = 0;
-			while(testingQue.length>0) {
-				let currStep = testingQue[0];
-				if (currStep[0] === endP[0] && currStep[1] === endP[1]) return true;
+  // the BFS algorithm that searches for a path between the 'start' and 'end' points
+  // returns 'true' if a path was found and 'false' if not
+  const pathExists = function () {
+    if (legalPoint(startP) && legalPoint(endP)) {
+      const testingQue = [ startP, ];
+      field[testingQue[0][0]][testingQue[0][1]].stepVisited = 0;
```

Comment:
Use destructuring to replace these numbers with something a human can understand.